Classes/System/TYPO3/Loader.php

```diff
@@ -25,156 +25,30 @@
  *  This copyright notice MUST APPEAR in all copies of the script!
  ***************************************************************/
 
-use TYPO3\CMS\Core\Core\Bootstrap;
-use TYPO3\CMS\Core\SingletonInterface;
-use TYPO3\CMS\Core\TimeTracker\NullTimeTracker;
-use TYPO3\CMS\Core\Utility\GeneralUtility;
 use TYPO3\CMS\Core\Authentication\BackendUserAuthentication;
+use TYPO3\CMS\Core\SingletonInterface;
 use TYPO3\CMS\Frontend\Authentication\FrontendUserAuthentication;
-use TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController;
-use TYPO3\CMS\Frontend\Utility\EidUtility;
-use LogicException;
 
 /**
+ * Starting TYPO 8 these dont need to be initializied anymore by our code because we use the psr-15 middleware layer concept
  * @package Restler
  */
 class Loader implements SingletonInterface
 {
-    /**
-     * defines, if usage of backend-user is enabled
-     *
-     * @var boolean
-     */
-    private $isBackEndUserInitialized = false;
-    /**
-     * defines, if usage of frontend-user is enabled (this is needed, if the eID-script must determine the frontend-user)
-     *
-     * @var boolean
-     */
-    private $isFrontEndUserInitialized = false;
-    /**
-     * defines, if frontend-rendering is enabled (this is needed, if the eID-script must render some content-elements or RTE-fields)
-     *
-     * @var boolean
-     */
-    private $isFrontEndRenderingInitialized = false;
-
     /**
      * @return BackendUserAuthentication
-     * @throws LogicException
      */
     public function getBackEndUser()
     {
-        if ($this->isBackEndUserInitialized === false) {
-            throw new LogicException('be-user is not initialized - initialize with BE-user with method \'initializeBackendEndUser\'');
-        }
         return $GLOBALS['BE_USER'];
     }
 
     /**
      * @return FrontendUserAuthentication
-     * @throws LogicException
      */
     public function getFrontEndUser()
     {
-        if ($this->isFrontEndUserInitialized === false) {
```

Comment:
Why don't we need this check anymore? We could end up returning null ?

Reply:
The Frontend user will be initialized by another middleware which should allways be executed before, so this should never happen.